src/ui/libs/DatalensChartkit/components/ChartKitBase/components/Header/components/Menu/Menu.js

```diff
@@ -199,29 +223,57 @@ export class Menu extends React.PureComponent {
         }
 
         const menuItems = this.prepareItems(visibleItems, data, onChange);
-        const ModalComponent = this.state?.modal;
+
+        const mobileItems = menuItems.flat();
```

Comment:
Can we traenform menuItems to flat array only if IS_MOBILE

Reply:
`menuItems.flat()` is moved inside the block with the `renderSheet`